pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go

```diff
@@ -382,64 +424,68 @@ func (s *Scheduler) removeIssuer(q *IssuerQueue, err error) {
 		s.events.BlockDropped.Trigger(block, err)
 	}
 
-	s.buffer.RemoveIssuer(q.IssuerID())
+	s.deficits.Delete(issuerID)
+
+	s.buffer.RemoveIssuer(issuerID)
 }
 
-func (s *Scheduler) getDeficit(accountID iotago.AccountID) (uint64, error) {
-	d, exists := s.deficits.Get(accountID)
-	if !exists {
-		_, err := s.manaRetrieveFunc(accountID)
-		// manaRetrieveFunc return error if mana is less than MinMana or the Mana can not be retrieved for the account.
-		if err != nil {
-			return 0, ierrors.Wrapf(err, "could not get deficit for issuer %s", accountID)
-		}
-		// load with max deficit if the issuer has Mana but has been removed from the deficits map
-		return s.optsMaxDeficit, nil
-	}
+func (s *Scheduler) createIssuer(accountID iotago.AccountID) *IssuerQueue {
+	issuerQueue := s.buffer.CreateIssuerQueue(accountID)
+	s.deficits.Set(accountID, 0)
 
-	return d, nil
+	return issuerQueue
 }
 
-func (s *Scheduler) updateDeficit(accountID iotago.AccountID, delta int64) error {
-	var err error
-	s.deficits.Compute(accountID, func(currentValue uint64, exists bool) uint64 {
-		if !exists {
-			_, err = s.manaRetrieveFunc(accountID)
-			// manaRetrieveFunc return error if mana is less than MinMana or the Mana can not be retrieved for the account.
-			if err != nil {
-				err = ierrors.Wrapf(err, "could not get deficit for issuer %s", accountID)
-				return 0
-			}
+// addExistingIssuer adds an existing issuer to the scheduler with Max deficit. This is for an optimisation where we remove max deficit issuers which have been inactive for some time.
+// func (s *Scheduler) addExistingIssuer(accountID iotago.AccountID) {
```

Comment:
Is this gonna be used?

Reply:
Not yet, so I suppose this should be removed.